### ISSUE TITLE
Name spacing conflict

### ISSUE THREAD

**Issue by bigearsenal** [hmlongco/Factory#161]
Hi there!
Thank you for an awesome library.
We faced a problem with the namespacing: 
- The library name is `Factory`, there is also a struct called `Factory<T>` in the library.
- We have a custom structure that is also named `Container`.
- We have tried to distinguish the Factory's Container by adding `Factory.Container.shared.myService1()` but it doesn't work since there is an `struct Factory<T>` that already took the Factory name.

In summary, we can not use the name `Container` for new structure anymore, since it conflicts with Factory's Container and we can not use `Factory.Container` when there is already a `struct Factory<T>` in the library. (The compiler is confused between `struct Factory<T>` and module name `Factory`>.

I suggest renaming the `Container` to `DIContainer`

**Comment by hmlongco** [comment]
The current version has an alias for Container, FactoryContainer

**Comment by bigearsenal** [comment]
Thank for answering, I saw that but they are right now not publicly visible.

**Comment by hmlongco** [comment]
derp

**Comment by bigearsenal** [comment]
See the commit. Thank you!
